File: LuckyBol/app/src/main/java/com/pplus/luckybol/core/network/upload/S3Upload.java
package com.pplus.luckybol.core.network.upload;

import com.google.gson.reflect.TypeToken;
import com.pplus.luckybol.Const;
import com.pplus.luckybol.core.network.model.dto.Attachment;
import com.pplus.luckybol.core.network.model.response.NewResultResponse;

/**
 * Created by j2n on 2016. 9. 30..
 */

public class S3Upload extends AbstractUpload<Attachment>{

    public S3Upload(PplusUploadListener<Attachment> pplusCallback){

        super(pplusCallback);
    }

    @Override
    String getUploadUrl(){

        return Const.API_URL + "common/saveOnlyS3";
    }

    @Override
    boolean isMultiThreadEnable(){

        return true;
    }

    @Override
    TypeToken<NewResultResponse<Attachment>> getResultType(){

        return new TypeToken<NewResultResponse<Attachment>>(){};
    }
}

File: WowBox/app/src/main/java/com/lejel/wowbox/core/network/upload/S3Upload.kt
package com.lejel.wowbox.core.network.upload

import com.google.gson.reflect.TypeToken
import com.lejel.wowbox.Const
import com.lejel.wowbox.core.network.model.response.NewResultResponse

/**
 * Created by j2n on 2016. 9. 30..
 */
class S3Upload(callback: PplusUploadListener<String>) : AbstractUpload<String>(callback) {


    override fun getUploadUrl(): String {
        return Const.API_URL + "file/s3Upload"
    }

    override fun isMultiThreadEnable(): Boolean {
        return true
    }

    override fun getResultType(): TypeToken<NewResultResponse<String>> {
        return object : TypeToken<NewResultResponse<String>>() {}
    }
}